aws-lc-rs/src/ed25519.rs

```diff
@@ -90,22 +92,56 @@ impl Debug for Ed25519KeyPair {
     }
 }
 
+impl Drop for Ed25519KeyPair {
+    fn drop(&mut self) {
+        self.private_key.zeroize();
+    }
+}
+
 #[derive(Clone)]
 #[allow(clippy::module_name_repetitions)]
-pub struct PublicKey {
-    public_key: [u8; ED25519_PUBLIC_KEY_LEN],
+/// The seed value for the `EdDSA` signature scheme using Curve25519
+pub struct Seed<'a>(&'a Ed25519KeyPair);
+
+/// Elliptic curve private key data encoded as a big-endian fixed-length integer.
+#[allow(clippy::module_name_repetitions)]
+pub struct Ed25519SeedBuffer {
+    _priv: (),
+}
+
+impl Seed<'_> {
+    /// Exposes the seed encoded as a big-endian fixed-length integer.
+    ///
+    /// For most use-cases, `EcdsaKeyPair::to_pkcs8()` should be preferred.
+    ///
+    /// # Errors
+    /// `error::Unspecified` if serialization failed.
+    pub fn to_buffer(&self) -> Result<Buffer<'static, Ed25519SeedBuffer>, Unspecified> {
+        let buffer = Vec::from(&self.0.private_key[..ED25519_PRIVATE_KEY_SEED_LEN]);
```

Comment:
Pre-existing, but the private_key should probably be wrapped in Zeroed and/or be heap allocated within the Ed25519KeyPair type. Probably worth filing an issue for cleanup in a future PR. But the buffer added here seems like it's doing the right thing.

Reply:
In the latest commit, the `Ed25519KeyPair` private key is in a Box to avoid the potential security concern.